sunpy/map/tests/test_map_factory.py

```diff
@@ -18,235 +18,226 @@
 import sunpy.data.test
 
 
-filepath = sunpy.data.test.rootdir
+filepath = pathlib.Path(sunpy.data.test.rootdir)
 a_list_of_many = [os.fspath(f) for f in pathlib.Path(filepath, "EIT").glob("*")]
 a_fname = a_list_of_many[0]
 
-AIA_171_IMAGE = os.path.join(filepath, 'aia_171_level1.fits')
-RHESSI_IMAGE = os.path.join(filepath, 'hsi_image_20101016_191218.fits')
-
-
-# ==============================================================================
-# Map Factory Tests
-# ==============================================================================
-class TestMap:
-    def test_mapsequence(self):
-        # Test making a MapSequence
-        sequence = sunpy.map.Map(a_list_of_many, sequence=True)
-        assert isinstance(sequence, sunpy.map.MapSequence)
-
-    def test_composite(self):
-        # Test making a CompositeMap
-        comp = sunpy.map.Map(AIA_171_IMAGE, RHESSI_IMAGE, composite=True)
-        assert isinstance(comp, sunpy.map.CompositeMap)
-
-    def test_patterns(self):
-        # Test different Map pattern matching
-
-        # File name
-        eitmap = sunpy.map.Map(a_fname)
-        assert isinstance(eitmap, sunpy.map.GenericMap)
-
-        # Directory
-        directory = pathlib.Path(filepath, "EIT")
+AIA_171_IMAGE = filepath / 'aia_171_level1.fits'
+RHESSI_IMAGE = filepath / 'hsi_image_20101016_191218.fits'
+
+amap = sunpy.map.Map(AIA_171_IMAGE)
+
+valid_map_inputs = [(a_fname, ),
```

Comment:
If `amap` is a fixture this would need to be a fixture, but then you can't use it in `parametrize` but that's probably not the end of the world looking at how you are using it.

Reply:
Am 👍 for making `amap` a fixture, but I had a trawl through the pytest docs and I can't work out how to make it play nicely with `parametrize`.

Reply:
I have tried a few things and gone down a few rabbit holes, and am struggling to work out a way to do this nicely, other than definitely a bunch of functions to return the map types, then calling the functions within the test (sort of like a manual fixture I guess).

Reply:
Personally I wouldn't waste too much time on it. We can always come back to this. I would rather we get this merged either in its current state or with more test cases than with more fixtures.

Reply:
In that case I think this is ready for review/merging